editor/editor_autoload_settings.cpp

```diff
@@ -419,6 +414,9 @@ Node *EditorAutoloadSettings::_create_autoload(const String &p_path) {
 
 		Ref<Script> scr = res;
 		if (scr.is_valid()) {
+			if (!scr->is_valid()) {
```

Comment:
This condition does not make sense.

Reply:
Strangely `.is_valid` and `->is_valid` does not valid the same thing. `.is_valid` checks if the resource exists and `->is_valid` checks if the script is valid (compiles). The problem I tried to fixe here is the wrong error message that was displayed when the script is invalid at startup: "Failed to create an autoload, script '%s' does not inherit from 'Node'" which is not necessarily true. Maybe there's a better way to do that?

Reply:
Ah yeah I didn't notice `.` and `->`, they are indeed different. So I guess this is correct.